test/crd.go

```diff
@@ -92,33 +92,41 @@ func BlueGreenRoute(namespace string, names, blue, green ResourceNames) *v1alpha
 	}
 }
 
-// Configuration returns a Configuration object in namespace with the name names.Config
-// that uses the image specified by imagePath.
-func Configuration(namespace string, names ResourceNames, imagePath string, options *Options) *v1alpha1.Configuration {
-	config := &v1alpha1.Configuration{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      names.Config,
-		},
-		Spec: v1alpha1.ConfigurationSpec{
-			RevisionTemplate: v1alpha1.RevisionTemplateSpec{
-				Spec: v1alpha1.RevisionSpec{
-					Container: corev1.Container{
-						Image:     imagePath,
-						Resources: options.ContainerResources,
-					},
-					ContainerConcurrency: v1alpha1.RevisionContainerConcurrencyType(options.ContainerConcurrency),
+// ConfigurationSpec returns the spec of a configuration to be used throughout different
+// CRD helpers.
+func ConfigurationSpec(imagePath string, options *Options) *v1alpha1.ConfigurationSpec {
+	spec := &v1alpha1.ConfigurationSpec{
+		RevisionTemplate: v1alpha1.RevisionTemplateSpec{
+			Spec: v1alpha1.RevisionSpec{
+				Container: corev1.Container{
+					Image:     imagePath,
+					Resources: options.ContainerResources,
 				},
+				ContainerConcurrency: v1alpha1.RevisionContainerConcurrencyType(options.ContainerConcurrency),
 			},
 		},
 	}
 
 	if options.RevisionTimeout > 0 {
-		config.Spec.RevisionTemplate.Spec.TimeoutSeconds = &metav1.Duration{Duration: options.RevisionTimeout}
+		spec.RevisionTemplate.Spec.TimeoutSeconds = &metav1.Duration{Duration: options.RevisionTimeout}
 	}
 
 	if options.EnvVars != nil && len(options.EnvVars) > 0 {
```

Comment:
[nit] do we need the len >0 check. Does it matter if I set envVars to []?

Reply:
Probably not, I'll take that out.